scrapy/link.py

```diff
@@ -28,6 +28,7 @@ def __init__(self, url, text='', fragment='', nofollow=False):
         self.text = text
         self.fragment = fragment
         self.nofollow = nofollow
+        self.encoding = encoding
```

Comment:
Tests seem to pass fine with `safe_url_string` in `LinkExtractor`. Checking further.